### PR TITLE
Add support for temporal point clouds

### PR DESCRIPTION
**Pull request by finsberg**
Adress https://github.com/scientificcomputing/scifem/issues/58

In my opinion we could remove the old code for writing point clouds and just use this instead. If so, we might want to add a deprecation warning. 

Current only writing is possible, but I think we could add support for read and append in the future.

### CODE REVIEW

File: tests/test_xdmf.py [hunk 1]
```diff
@@ -2,12 +2,78 @@
...
+            xdmf.write(t.value)
+
+            t.value = 0.3
+            u.interpolate(dolfinx.fem.Expression(u_expr, V.element.interpolation_points()))
+            v.interpolate(dolfinx.fem.Expression(v_expr, V.element.interpolation_points()))
+            xdmf.write(t.value)
+    else:
+        xdmf = scifem.xdmf.XDMFFile(folder / "data.xdmf", [u, v], backend=backend)
+
+        # breakpoint()
```
Comment (jorgensd) [inline tests/test_xdmf.py:61]:
```suggestion
```

File: tests/test_xdmf.py [hunk 2]
```diff
@@ -2,12 +2,78 @@
...
+    else:
+        u_expr = X * ufl.cos(ufl.pi * t)
+        v_expr = X * ufl.sin(X[0]) * ufl.cos(ufl.pi * t)
+
+    u.name = "u"
+    v.name = "v"
+
+    from pathlib import Path
+
+    folder = Path(".")
```
Comment (jorgensd) [inline tests/test_xdmf.py:46]: Tmppath?

Comment (finsberg) [reply tests/test_xdmf.py:46]: Whoops😅 Will fix

File: src/scifem/xdmf.py [hunk 3]
```diff
@@ -338,3 +339,223 @@ def create_pointcloud(
...
+        ]
+        self.xdmfname.write_text("".join(text))
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.close()
+
+    def write(self, time: float) -> None:
```
Comment (jorgensd) [inline src/scifem/xdmf.py:555]: Docstring?

**Comment by finsberg** [comment]
Still need to implement "read" and "append" methods but I think this could be a "Good first issue" for new contributors

**Pull request merged**